bftengine/tests/testSerialization/TestSerialization.cpp

```diff
@@ -148,7 +148,7 @@ const uint16_t msgsNum = 2 * fVal + 2 * cVal + 1;
 
 typedef pair<uint16_t, string> IdToKeyPair;
 
-ReplicaConfig config;
+ReplicaConfig *config;
```

Comment:
Why dynamically allocate? I also don't see a corresponding delete. Better to use some RAII, so we don't have leaks, no?

Reply:
because ReplicaConfigSingleton holds a pointer to ReplicaConfig

Reply:
I have changed an approach a little. Now ReplicaConfigSingleton doesn't hold a pointer to the original ReplicaConfig, but creates a copy, decoupling the life cycles of two objects